modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/upgrade/v3_0_0/UpgradePortletId.java

```diff
@@ -84,8 +85,8 @@ protected String[][] getRenamePortletIdsArray(
 					"ClientExtensionEntry")) {
 
 			while (resultSet.next()) {
-				String externalReferenceCode = resultSet.getString(
-					"externalReferenceCode");
+				String externalReferenceCode = _formatExternalReferenceCode(
```

Comment:
I believe you can use `PortalUtil.getJsSafePortletId()` method to do this conversion instead of using your own. That should be the canonical way. However, I don't see many usages of `PortalUtil.getJsSafePortletId()` in portal, so I'm not 100% sure...

Can you please check if using that method works for this particular bug and, if so, use it instead. Otherwise, it's possible that Brian asks you to do so, too.

Thx :-)

Reply:
Hi @izaera, `PortalUtil.getJsSafePortletId()` is intended for removing specific characters like dash, period, and space in the name. However, this fix is trying to update the portletId that was deployed with the format like `com_liferay_client_extension_web_internal_portlet_ClientExtensionEntryPortlet_xxx_xxx_xxx`.

Reply:
In that case, go ahead. You know better than me what needs to be done :-)

Reply:
I'll forward the PR to Brian.